src/test/java/uk/gov/pay/directdebit/payers/dao/PayerDaoIT.java

```diff
@@ -173,4 +173,57 @@ public void shouldGetAPayerByPaymentRequestId() {
     public void shouldNotFindAPayerByRequestId_ifIdIsInvalid() {
         assertThat(payerDao.findByPaymentRequestId(154L).isPresent(), is(false));
     }
+
+    @Test
+    public void shouldUpdatePayerDetailsAndReturnNumberOfAffectedRows() {
+        testPayer.insert(testContext.getJdbi());
+
+        String newName = "coffee";
+        String newAccountNumber = "87654321";
+        String newSortCode = "123456";
+        String newEmail = "aaa@bbb.test";
+        boolean newRequiresAuthorisation = true;
+        Payer newPayerDetails = PayerFixture
+                .aPayerFixture()
+                .withName(newName)
+                .withAccountNumber(newAccountNumber)
+                .withSortCode(newSortCode)
+                .withEmail(newEmail)
+                .withAccountRequiresAuthorisation(newRequiresAuthorisation)
+                .withAccountNumberLastTwoDigits("21")
+                .withAddressLine1(ADDRESS_LINE_1)
+                .withAddressLine2(ADDRESS_LINE_2)
+                .withAddressCity(ADDRESS_CITY)
+                .withAddressPostcode(ADDRESS_POSTCODE)
+                .withAddressCountry(ADDRESS_COUNTRY)
+                .withExternalId("this_should_not_be_updated")
+                .withCreatedDate(ZonedDateTime.now())
+                .toEntity();
+        int numOfUpdatedPayers = payerDao.updatePayerDetails(testPayer.getId(), newPayerDetails);
+        Map<String, Object> payerAfterUpdate = testContext.getDatabaseTestHelper().getPayerById(testPayer.getId());
+
+        assertThat(numOfUpdatedPayers, is(1));
+        assertThat(payerAfterUpdate.get("id"), is(testPayer.getId()));
+        assertThat(payerAfterUpdate.get("payment_request_id"), is(testPaymentRequest.getId()));
+        assertThat(payerAfterUpdate.get("external_id"), is(EXTERNAL_ID)); //old external id
```

Comment:
Maybe group everything that shouldn’t be updated together at the bottom introduced with a comment like `// These properties cannot be updated`?

Reply:
👍